internal/rbd/replication.go

```diff
@@ -19,31 +19,31 @@ package rbd
 import (
 	"context"
 	"strings"
+	"fmt"
 
 	librbd "github.com/ceph/go-ceph/rbd"
 	"github.com/csi-addons/spec/lib/go/replication"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
 func (rv *rbdVolume) ResyncVol(localStatus librbd.SiteMirrorImageStatus, force bool) error {
 	if resyncRequired(localStatus) {
 		// If the force option is not set return the error message to retry
 		// with Force option.
 		if !force {
-			return status.Errorf(codes.FailedPrecondition,
-				"image is in %q state, description (%s). Force resync to recover volume",
-				localStatus.State, localStatus.Description)
+		 	fmt.Errorf("image is in %q state, description (%s). Force resync to recover volume", localStatus.State, localStatus.Description)
```

Comment:
These `fmt.Errorf(....)` statements return an error, and a lot of them are not used/returned.

You can probably do something like this which will be better:

```go
return fmt.Errorf("%w: image is in %q state, description (%s). Force resync to recover volume", ErrFailedPrecondition, localStatus.State, localStatus.Description)
```
or
```go
err = fmt.Errorf("%w: image is in %q state, description (%s). Force resync to recover volume", ErrFailedPrecondition, localStatus.State, localStatus.Description)
return err
```
By using `%w` and pasing `ErrFailedPrecondition` for the formatting, you get the error message in the error string, and the type of the error will still be `ErrFailedPrecondition`.

Reply:
Yes Niels, I have already done this here (in 2nd commit), 			
return fmt.Errorf("image is in %q state, description (%s). Force resync to recover volume: %w",